Switch watchtower to hawk

Kynde opened the pull request:
This is an example PR to replace the simple _watchtower_ with something that uses the java file system watcher nio api which in turn utlilizes the os fs watching mechanisms, e.g. _inotify_ on linux.

The java file system watcher api unfortunately stuffs most of the inotify events under one "modified" umbrella and as a result a mere vim file save produces three modified events per file. I opted to add a simple timeout burst filter just to see if it would work as before.

There's room for improvements and clean up, but I didn't want to go there until I know whether there was some specific reason to use watchtower or not.

signed-off-by: Tommi Kyntölä tommi.kyntola@gmail.com

Kynde committed:
Switch watchtower to hawk

Deraen commented on src/leiningen/bat_test.clj:
```diff
@@ -43,24 +43,28 @@
+                               :filter hawk.core/file?
+                               :context (constantly 0)
+                               :handler (fn [~'ctx ~'e]
+                                          (if (and (clojure.string/ends-with? (.getName (:file ~'e)) ".clj")
```
Should also look for `.cljc` files.

Kynde replied:
Sure. Done.

Deraen commented on src/leiningen/bat_test.clj:
```diff
@@ -43,24 +43,28 @@
+                               :context (constantly 0)
+                               :handler (fn [~'ctx ~'e]
+                                          (if (and (clojure.string/ends-with? (.getName (:file ~'e)) ".clj")
+                                                   (= :modify (:kind ~'e))
```
Maybe no need to filter by event kind, because new file should also trigger tests?

And to match previous code, should ignore events where filename starts with dot. As some editor could create hidden temp files with same extension.

Kynde replied:
Agreed. Done.

Kynde committed:
Match earlier watchtower behaviour more closely

Review comments addressed. Include .cljc files to watch and ignore dot-files.
Also now diregards event type.
